display/source_callbacks_generator.rb

```diff
@@ -0,0 +1,52 @@
+require 'yaml'
+require 'erb'
+
+DOCS = <<-DOCS
+Command usage: ruby source_callbacks_generator.rb <callbacks_instance.yaml> <callbacks_source_file.c>
+
+    Generates SOURCE component callbacks to push messages downtream as defined in the callbacks_instance.yaml
+
+    Example: 
+        ruby source_callbacks_generator.rb ./3.interval_instances.yaml ./SOURCE.interval/interval_callbacks.c
+
+DOCS
+
+# Testing minimun arguments requirement
+if ARGV.count != 2 then 
+    puts DOCS
+    return
+end
+
+test_file_path = ARGV[0]
+out_file_path = ARGV[1]
+working_path = File.expand_path(File.dirname(__FILE__))
+
+data = YAML.load_file(test_file_path)
+entries = data[:entries]
+
+template =  <<-TEXT
+#include "component.h"
+#include "create.h"
+#include <stdio.h>
+
```

Comment:
I think we need to add the definition of an empty initialize, finalize function

Reply:
Maybe..., we can discuss in which cases the user need to initialize or finalize, since the only thing he has to do is push messages downstream and the source component does not call  these functions yet.

Reply:
As metababel is concerned, I think it make sense to always required it.
It's user responsibility to always provide it. In the case of the tally tests-source, we should just generate an empty definition.